### ISSUE TITLE
bug(sqlsmith): queries still not reproducible in madsim

### ISSUE THREAD

**Issue by kwannoel** [risingwavelabs/risingwave#8152]
### Describe the bug

After #8068 , it was expected to sqlsmith should be able to reproduce results with each run.

The sequence of rng generated is still reproducible see:
[local.txt](https://github.com/risingwavelabs/risingwave/files/10812180/local.txt)
[out.txt](https://github.com/risingwavelabs/risingwave/files/10812181/out.txt)

However it does not yield the same queries.

I suspect this is because the sequence of query-gen can be out of order due to async execution.

Simple fix can be to write a synchronous program, by refactoring to use postgres_client.

This pr: https://github.com/risingwavelabs/risingwave/pull/7967 will help to stabilize `main` and `PR` workflow in the meantime.

**Comment by kwannoel** [comment]
cc @wangrunji0408

**Comment by wangrunji0408** [comment]
Yes, I also think async execution is the reason for irreproducible. If we could make all queries be generated sequentially at first, this problem might be resolved. BTW, we can't use postgres_client because it is not adapted to the simulator and will block the thread which is not allowed in the simulation. 🥲
